### PR TITLE
test: fix nit after #2591

### PR DESCRIPTION
**Pull request by dio**
*test: fix nit after #2591*

*Description*:
Fix some nit for PR #2591.

*Risk Level*: Low

*Testing*:
unit

*Docs Changes*:
N/A

*Release Notes*:
N/A

### CODE REVIEW

File: test/common/upstream/health_checker_impl_test.cc [hunk 1]
```diff
@@ -495,7 +495,7 @@ TEST_F(HttpHealthCheckerImplTest, SuccessServiceCheckWithCustomHostValue) {
         EXPECT_TRUE(headers.Path());
         EXPECT_NE(nullptr, headers.Host());
         EXPECT_NE(nullptr, headers.Path());
-        EXPECT_EQ(headers.Host()->value().c_str(), std::string(host));
+        EXPECT_EQ(headers.Host()->value().c_str(), host);
         EXPECT_EQ(headers.Path()->value().c_str(), std::string("/healthcheck"));
```
Comment (htuch) [inline test/common/upstream/health_checker_impl_test.cc]: Why not this line as well?

**Comment by dnoe** [comment]
Looks like the TSAN failure is a flake, I am investigating that. When you address @htuch's comment and push a new commit it will re-run the test.

File: test/common/upstream/health_checker_impl_test.cc (hunk 1 above)
Comment (dio) [reply test/common/upstream/health_checker_impl_test.cc]: Ah, OK. BTW @htuch do you think it is worth adding more tests as V2 YAML in this test file?

Comment (htuch) [reply test/common/upstream/health_checker_impl_test.cc]: Generally, moving as much as possible to v2 YAML will be a good thing. I think the best model to do it is organically, e.g. as we go and modify or add new tests, but any cleanup is always appreciated.

**Comment by htuch** [comment]
Please merge master to pick up https://github.com/envoyproxy/envoy/pull/2613, this should fix CI TSAN.

**Comment by dio** [comment]
It seems fine now 🙌

**Pull request merged**